src/axom/core/ArrayBase.hpp

```diff
@@ -450,21 +470,24 @@ class ArrayBase
    */
   AXOM_HOST_DEVICE IndexType memorySize() const
   {
-    IndexType maxSize = 0;
-    for(int dim = 0; dim < DIM; dim++)
-    {
-      maxSize = axom::utilities::max(maxSize, m_strides[dim] * m_shape[dim]);
-    }
-    return maxSize;
+    auto slowestDir = m_indexer.slowestDirs()[0];
+    return m_indexer.strides()[slowestDir] * m_shape[slowestDir];
   }
 
-  /// \brief Memory offset to a slice at the given lower-dimensional index.
+  /*!
+    \brief Memory offset to a slice at the given lower-dimensional index.
+
+    Allowed only for row-major arrays.
+
+    @pre indexer().getStrideOrder() & ArrayStrideOrder::ROW == true
+  */
   template <int UDim>
   AXOM_HOST_DEVICE IndexType offset(const StackArray<IndexType, UDim>& idx) const
   {
     static_assert(UDim <= DIM,
                   "Index dimensions cannot be larger than array dimensions");
-    return numerics::dot_product(idx.begin(), m_strides.begin(), UDim);
+    assert(indexer().getStrideOrder() & ArrayStrideOrder::ROW);
```

Comment:
If `getStrideOrder()` isn't a bitmask, you might wanna just use `==` to clarify intent.

Reply:
It *is* a bitmask.